packages/core/src/codewhisperer/util/zipUtil.ts

```diff
@@ -0,0 +1,129 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import admZip from 'adm-zip'
+import * as vscode from 'vscode'
+import path from 'path'
+import { tempDirPath } from '../../shared/filesystemUtilities'
+import { getLogger } from '../../shared/logger'
+import * as CodeWhispererConstants from '../models/constants'
+import { fsCommon } from '../../srcShared/fs'
+import { statSync } from 'fs'
+
+export interface ZipMetadata {
+    rootDir: string
+    zipFilePath: string
+    scannedFiles: Set<string>
+    srcPayloadSizeInBytes: number
+    buildPayloadSizeInBytes: number
+    zipFileSizeInBytes: number
+    lines: number
+}
+
+export const ZipConstants = {
+    newlineRegex: /\r?\n/,
+}
+
+export class ZipUtil {
```

Comment:
Is this really doing much that is codewhisperer specific? Things like `fileScanPayloadSizeLimitBytes` could be parameterized. If a "zip util" is needed then presumably that would be useful to other parts of the codebase, not only CW.

Reply:
There will be more changes included in the next PR for project scans. Will look into making it generic in a separate PR